### PR TITLE
ICU-21896 Use double base in pow()

### PR DESCRIPTION
**Pull request by mojca**
gcc 5.5 on Solaris refuses to recognise pow(int, int32_t)

(I would reword the commit message, but I need some hints)

##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21896
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)

Potentially related to #1875.

### CODE REVIEW

**Comment by mojca** [comment]
@echeran

File: icu4c/source/i18n/plurrule.cpp [hunk 1]
```diff
@@ -1618,7 +1618,7 @@ FixedDecimal::FixedDecimal(double n, int32_t v, int64_t f, int32_t e) {
     init(n, v, f, e);
     // check values. TODO make into unit test.
     //            
-    //            long visiblePower = (int) Math.pow(10, v);
+    //            long visiblePower = (int) Math.pow(10.0, v);
```
Comment (srl295) [inline icu4c/source/i18n/plurrule.cpp:1621]: this is a comment

**Comment by srl295** [comment]
@mojca commit msg is fine, just needs accepted ticket

File: icu4c/source/i18n/plurrule.cpp (hunk 1 above)
Comment (mojca) [reply icu4c/source/i18n/plurrule.cpp:1621]: I know that it doesn't make any difference, but it makes sense to keep the code consistent even in the comments in case one day someone decides to uncomment those lines again. That said, if you prefer this not to be patched, I can remove it from the PR.

**Comment by sffc** [comment]
@echeran owns some tickets to work on this part of the plural rules code and is the right person to review this.

**Review by echeran — APPROVED** [review]
LGTM
